Modules/Loadable/Colors/Logic/vtkSlicerColorLogic.cxx

```diff
@@ -209,3 +214,38 @@ std::vector<std::string> vtkSlicerColorLogic::FindColorFiles(const std::vector<s
     } // end of looping over dirs
   return filenames;
 }
+
+//----------------------------------------------------------------------------
+vtkMRMLColorBarDisplayNode* vtkSlicerColorLogic::CreateAndObserveColorBarForNode(vtkMRMLDisplayableNode* dispNode)
+{
+  if (!dispNode)
+  {
+    vtkDebugMacro("CreateAndObserveColorBarForNode: Displayable node is invalid");
+    return nullptr;
+  }
+
+  vtkMRMLColorBarDisplayNode* cbNode = vtkMRMLColorBarDisplayNode::SafeDownCast(
+    dispNode->GetNodeReference(vtkMRMLColorBarDisplayNode::COLOR_BAR_REFERENCE_ROLE));
+  if (cbNode)
+  {
+    vtkDebugMacro("CreateAndObserveColorBarForNode: Displayable node already has color bar node");
+    return cbNode;
+  }
+
+  vtkMRMLScene* mrmlScene = this->GetMRMLScene();
+  if (!mrmlScene)
+  {
+    vtkErrorMacro("CreateAndObserveColorBarForNode: Invalid MRML scene");
+    return nullptr;
+  }
+
+  // Create color bar and set reference
+  vtkNew<vtkMRMLColorBarDisplayNode> colorBarNode;
```

Comment:
Use scene->AddNewNodeByClass instead of vtkNew+AddNode, because that is shorter and it also allows user to override default node properties (using the scene's default node mechanism).